openqa-prepare-mm-setup

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env perl
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+use strict;
+use autodie ':all';
+
+system("zypper in -y os-autoinst-openvswitch");
+system("systemctl enable --now os-autoinst-openvswitch");
+
+my $bridge = 1;
+my $openvswitch = "/etc/sysconfig/os-autoinst-openvswitch";
+open my $fh, '>', $openvswitch or die "Failed to open $openvswitch for writing!";
+print $fh "OS_AUTOINST_USE_BRIDGE=br$bridge";
+close($fh);
+
+system("ovs-vsctl add-br br$bridge");
+print("Added bridge $bridge");
+
+my $bridge_file = "/etc/sysconfig/network/ifcfg-br$bridge";
+my $ip = "10.0.2.2/15";
+my $config = "
+BOOTPROTO='static'                                                                                                         
+IPADDR='$ip'                                                                                                       
+STARTMODE='auto'                                                                                                           
+ZONE=trusted                                                                                                               
+OVS_BRIDGE='yes'                                                                                                           
+PRE_UP_SCRIPT=\"wicked:gre_tunnel_preup.sh\"
+";
+
+for my $i (0..147) {
+    $config .= "OVS_BRIDGE_PORT_DEVICE_$i='tap$i'\n";
+
+    my $tapconfig = "
+    BOOTPROTO='none'
+    IPADDR=''
+    NETMASK=''
+    PREFIXLEN=''
+    STARTMODE='auto'
+    TUNNEL='tap'
+    TUNNEL_SET_GROUP='nogroup'
+    TUNNEL_SET_OWNER='_openqa-worker'
+    ZONE=trusted
+    ";
+
+    my $tap_file = "/etc/sysconfig/network/ifcfg-tap$i";
+    open my $fh, '>', $tap_file or die "Failed to open $tap_file for writing!";
+    print $fh $tapconfig;
+    close($fh);
+}
+
+open my $fh, '>', $bridge_file or die "Failed to open $bridge_file for writing!";
+print $fh $config;
+close($fh);
+print("Config written!");
+
+my $gre = "/etc/wicked/scripts/gre_tunnel_preup.sh";
+open my $fh, '>', $gre or die "Failed to open $gre for writing!";
+
+my $gre_config = '
+#!/bin/sh
+action="$1"
+bridge="$2"
+ovs-vsctl set bridge $bridge stp_enable=true
+';
+
+my %workers = (openqaworker1 => '192.168.112.6',
```

Comment:
It would be nice if that mapping was read from a JSON/Yaml file.

Reply:
Yeah. I was actually hoping we could get rid of IP's and simply specify those via the command line. Because I'm not sure where such a file would live. And we'll have to maintain it.

Reply:
It could live in *some* repo. We definitely need some kind of definition somewhere. Making this information up each time from scratch and passing it as CLI args each time when the script is executed seems not like a good idea.

Reply:
I wouldn't expand that setting at all. By default no one should care to handle IPv4 addresses manually. Why not just call `nslookup | sed ...` to get the IPv4 address?